README.md
# date_time_picker_tr

A new Flutter package project.

## Screenshots
<img src="https://user-images.githubusercontent.com/37551474/114924556-c9714b80-9e36-11eb-8284-2dfe02f5b7ac.png" height="300" />

## Usage

[Example](https://github.com/taylanyildiz/date_time_picker_tr/tree/main/example/example.dart)

To use this package :

* add the dependency to your [pubspec.yaml](https://github.com/taylanyildiz/date_time_picker_tr/blob/main/pubspec.yaml) file.

```yaml
  dependencies:
    flutter:
      sdk: flutter
    date_time_picker_tr:
```

### How to use

```dart
/*
    Here we have minDate , maxDate and initialDate
*/
DatePicker(
    maxDate: DateTime.parse('2020-10-13'),
    minDate: DateTime.parse('1980-10-13'),
    initialDate: DateTime.parse('2019-09-30'),
    onChangedDate: (dateTime) => print(dateTime),
    overlayColor: Colors.grey,
    backgroundColor: Colors.black,
    textColor: Colors.white,
    height: 300.0,
    width: 100.0,
),
```


## Getting Started

This project is a starting point for a Dart
[package](https://flutter.dev/developing-packages/),
a library module containing code that can be shared easily across
multiple Flutter or Dart projects.

For help getting started with Flutter, view our 
[online documentation](https://flutter.dev/docs), which offers tutorials, 
samples, guidance on mobile development, and a full API reference.


CHANGELOG.md
## [0.0.1] - Relase 

## [0.1.1] - Date Time Picker on desktop tr language
## [0.1.2] - Date Time Picker on desktop tr language
## [0.1.3] - Date Time Picker on desktop tr language
## [0.1.4] - Edit Photos

lib/date_time_picker_tr.dart
library date_time_picker_tr;

import 'package:flutter/cupertino.dart';
import 'package:flutter/material.dart';

// ignore: must_be_immutable
class DatePicker extends StatefulWidget {
  DatePicker({
    Key key,
    @required this.initialDate,
    @required this.minDate,
    @required this.maxDate,
    @required this.onChangedDate,
    double height,
    double width,
    Color textColor,
    Color backgroundColor,
    Color overlayColor,
  })  : assert(initialDate.isAfter(minDate)),
        height = height ?? 300.0,
        width = width ?? 100.0,
        textColor = textColor ?? Colors.black,
        backgroundColor = backgroundColor ?? Colors.white,
        overlayColor = overlayColor ?? Colors.green,
        super(key: key);
  final DateTime initialDate;
  final DateTime minDate;
  final DateTime maxDate;
  final Function(DateTime) onChangedDate;
  double height;
  double width;
  Color textColor;
  Color backgroundColor;
  Color overlayColor;

  @override
  _DatePickerState createState() => _DatePickerState();
}

class _DatePickerState extends State<DatePicker> {
  final listMoths = <String>[
    "Ocak",
    "Şubat",
    "Mart",
    "Nisan",
    "Mayıs",
    "Haziran",
    "Temmuz",
    "Ağustos",
    "Eylül",
    "Ekim",
    "Kasım",
    "Aralık"
  ];

  int _day;
  int _month;
  int _year;

  _changeDay(int day) {
    _day = day + 1;
    dateTime = DateTime(_year, _month, _day);
    widget.onChangedDate(dateTime);
  }

  _changeMoth(int month) {
    _month = month + 1;
    dateTime = DateTime(_year, _month, _day);
    widget.onChangedDate(dateTime);
  }

  _changeYear(int year) {
    _year = year + widget.minDate.year;
    dateTime = DateTime(_year, _month, _day);
    widget.onChangedDate(dateTime);
  }

  //Day , Month , Year Settings
  _days() {
    var listWidgetDay = <Widget>[];
    for (var i = 1; i <= 31; i++) {
      listWidgetDay.add(Center(
          child: Text(
        '${i}',
        style: TextStyle(
          color: widget.textColor,
          fontWeight: FontWeight.bold,
        ),
      )));
    }
    return listWidgetDay;
  }

  _months() {
    var listWidgetMoth = <Widget>[];
    for (var i = 0; i < DateTime.monthsPerYear; i++) {
      listWidgetMoth.add(Center(
          child: Text(
        listMoths[i],
        style: TextStyle(
          color: widget.textColor,
          fontWeight: FontWeight.bold,
        ),
      )));
    }
    return listWidgetMoth;
  }

  _years() {
    var listWidgetYear = <Widget>[];
    for (var i = widget.minDate.year; i <= widget.maxDate.year; i++) {
      listWidgetYear.add(Center(
          child: Text(
        '${i}',
        style: TextStyle(
          color: widget.textColor,
          fontWeight: FontWeight.bold,
        ),
      )));
    }
    return listWidgetYear;
  }

  //Setting date
  FixedExtentScrollController _controllerDay;
  FixedExtentScrollController _controllerMoth;
  FixedExtentScrollController _controllerYear;
  DateTime dateTime;
  @override
  void initState() {
    // TODO: implement initState
    super.initState();

    _controllerDay =
        FixedExtentScrollController(initialItem: widget.initialDate.day - 1);

    _controllerMoth =
        FixedExtentScrollController(initialItem: widget.initialDate.month - 1);

    _controllerYear = FixedExtentScrollController(
        initialItem: widget.initialDate.year - widget.minDate.year);
    //
    dateTime = DateTime.parse(widget.initialDate.toString());
    _day = dateTime.day;
    _month = dateTime.month;
    _year = dateTime.year;
  }

  @override
  void dispose() {
    // TODO: implement dispose
    super.dispose();
    _controllerDay.dispose();
    _controllerMoth.dispose();
    _controllerYear.dispose();
  }

  @override
  Widget build(BuildContext context) {
    return Stack(
      alignment: Alignment.center,
      children: [
        Container(
          color: widget.backgroundColor,
          child: Row(
            mainAxisAlignment: MainAxisAlignment.center,
            children: [
              Container(
                height: widget.height,
                width: widget.width,
                child: CupertinoPicker(
                  backgroundColor: widget.backgroundColor,
                  scrollController: _controllerDay,
                  selectionOverlay: null,
                  looping: true,
                  itemExtent: 50.0,
                  onSelectedItemChanged: _changeDay,
                  children: _days(),
                ),
              ),
              Container(
                height: widget.height,
                width: widget.width,
                child: CupertinoPicker(
                  backgroundColor: widget.backgroundColor,
                  scrollController: _controllerMoth,
                  selectionOverlay: null,
                  looping: true,
                  itemExtent: 50.0,
                  onSelectedItemChanged: _changeMoth,
                  children: _months(),
                ),
              ),
              Container(
                height: widget.height,
                width: widget.width,
                child: CupertinoPicker(
                  backgroundColor: widget.backgroundColor,
                  scrollController: _controllerYear,
                  selectionOverlay: null,
                  looping: true,
                  itemExtent: 50.0,
                  onSelectedItemChanged: _changeYear,
                  children: _years(),
                ),
              ),
            ],
          ),
        ),
        IgnorePointer(
          child: Container(
            width: 350.0,
            height: 50,
            decoration: BoxDecoration(
              color: widget.overlayColor.withOpacity(.4),
              borderRadius: BorderRadius.circular(10.0),
            ),
          ),
        ),
      ],
    );
  }
}


example/example.dart
import 'package:date_time_picker_tr/date_time_picker_tr.dart';
import 'package:flutter/cupertino.dart';
import 'package:flutter/material.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      debugShowCheckedModeBanner: false,
      title: 'DateTime Picker Desktop',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: HomeScreen(
        title: 'DateTime Picker Desktop',
      ),
    );
  }
}

class HomeScreen extends StatefulWidget {
  final String title;
  HomeScreen({
    Key key,
    this.title,
  }) : super(key: key);
  @override
  _HomeScreenState createState() => _HomeScreenState();
}

class _HomeScreenState extends State<HomeScreen> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(),
      body: Center(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: [
            DatePicker(
              maxDate: DateTime.parse('2020-10-13'),
              minDate: DateTime.parse('1980-10-13'),
              initialDate: DateTime.parse('2019-09-30'),
              onChangedDate: (dateTime) => print(dateTime),
              overlayColor: Colors.grey,
              backgroundColor: Colors.black,
              textColor: Colors.white,
              height: 300.0,
              width: 100.0,
            ),
          ],
        ),
      ),
    );
  }
}
